src/item/linkItem/linkItem.test.ts

```diff
@@ -0,0 +1,48 @@
+import { describe, expect, it } from 'vitest';
+
+import { ItemType } from '../itemType.js';
+import { buildLinkExtra, getLinkThumbnailUrl } from './linkItem.js';
+
+describe('getLinkThumbnailUrl', () => {
+  it('returns thumbnail over icon', () => {
```

Comment:
I am not sure we want to return the thumbnails over the icons in all situations. 

From my understanding and experimental research, 
Icons: usually the website favicon, when multiple are provided they are ranked by order or biggest to smallest (favicon.ico is usually last).
Thumbnails: when the site provides a social preview, this is where we can get it. They are usually sized like banners. 

Below you will find a few examples:

<img width="1470" alt="Screenshot 2024-07-15 at 16 55 39" src="https://github.com/user-attachments/assets/4730d47d-dd2c-43eb-bd94-c0ea7c7199c3">
<img width="1470" alt="Screenshot 2024-07-15 at 16 55 29" src="https://github.com/user-attachments/assets/62c62feb-f6be-4213-af14-0fa845b0c53e">
<img width="1470" alt="Screenshot 2024-07-15 at 16 55 21" src="https://github.com/user-attachments/assets/c17d18d7-778b-4acc-bbe9-a8c5ee6569b9">

So I feel like this really depends on two things: size and aspect ratio.

At smaller sizes I would prefer to render the icons. 
At square aspect ratios the icons are usually better since they are required to fit in a square space.
At larger sizes and without contraint on aspect ratio, thumbnails are better as they can contain more information related to the context.

Reply:
I think what Basile said is really interesting !